packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -638,8 +617,42 @@ class PaddleSliderValueIndicatorShape extends SliderComponentShape {
     return _bottomLobeEnd;
   }
 
-  void _drawValueIndicator(Canvas canvas, Offset center, Paint paint, double scale,
-      TextPainter labelPainter, double textScaleFactor) {
+  // Determines the "best" offset to keep the bubble on the screen. The calling
+  // code will bound that with the available movement in the paddle shape.
+  double _getIdealOffset(RenderBox parentBox, double halfWidthNeeded, double scale, Offset center) {
+    const double edgeMargin = 4.0;
+    final Rect topLobeRect = new Rect.fromLTWH(
+      -_topLobeRadius - halfWidthNeeded,
+      -_topLobeRadius - _distanceBetweenTopBottomCenters,
+      2.0 * (_topLobeRadius + halfWidthNeeded),
+      2.0 * _topLobeRadius,
+    );
+    final Offset origin = center - parentBox.localToGlobal(Offset.zero);
+    // We can just multiply by scale instead of a transform, since we're scaling
+    // around (0, 0).
+    final Offset topLeft = parentBox.localToGlobal((topLobeRect.topLeft * scale) + origin);
+    final Offset bottomRight = parentBox.localToGlobal((topLobeRect.bottomRight * scale) + origin);
+    final Size screenBounds = window.physicalSize / window.devicePixelRatio;
```

Comment:
Shouldn't this be based on MediaQueryData? What if a slider's ancestor has overridden MediaQuery?

Reply:
Hmm.  I suppose it could be, but I really want the physical screen, not a virtual one.  I only want to avoid the sides of a physical screen.  You're thinking that if someone changes the device pixel ratio that this should honor that (because otherwise it won't do what I want)?

Reply:
Actually, in thinking more about this, I realized that you're right.  For instance, on platforms where there might be multiple "Windows", this might be crucial.